artifactory/utils/buildinfoproperties_test.go

```diff
@@ -63,21 +70,55 @@ func TestCreateSimplePropertiesFileWithProxy(t *testing.T) {
 	}
 	createSimplePropertiesFile(t, propertiesFileConfig)
 	setProxy(proxyOrg, t)
+	setNoProxy(noproxyOrg, t)
 }
 
-func TestCreateSimplePropertiesFileWithoutProxy(t *testing.T) {
+func TestCreateSimplePropertiesFileWithoutProxyAndNoProxy(t *testing.T) {
 	proxyOrg := getOriginalProxyValue()
+	noproxyOrg := getOriginalNoProxyValue()
 	setProxy("", t)
+	setNoProxy("", t)
 	var propertiesFileConfig = map[string]string{
 		"artifactory.resolve.contextUrl": "http://some.url.com",
 		"artifactory.publish.contextUrl": "http://some.other.url.com",
 		"artifactory.deploy.build.name":  "buildName",
 	}
 	createSimplePropertiesFile(t, propertiesFileConfig)
 	setProxy(proxyOrg, t)
+	setNoProxy(noproxyOrg, t)
 
 }
 
+func TestCreateSimplePropertiesFileWithProxyAndNoProxy(t *testing.T) {
+	proxyOrg := getOriginalProxyValue()
+	noproxyOrg := getOriginalNoProxyValue()
+	setProxy(proxy, t)
+	setNoProxy(noproxy, t)
+	var propertiesFileConfig = map[string]string{
+		"artifactory.resolve.contextUrl": "http://some.url.com",
+		"artifactory.publish.contextUrl": "http://some.other.url.com",
+		"artifactory.deploy.build.name":  "buildName",
+	}
+	createSimplePropertiesFile(t, propertiesFileConfig)
+	setProxy(proxyOrg, t)
+	setNoProxy(noproxyOrg, t)
+}
+
+func TestCreateSimplePropertiesFileWithProxyAndDifferentNoProxy(t *testing.T) {
+	proxyOrg := getOriginalProxyValue()
+	noproxyOrg := getOriginalNoProxyValue()
+	setProxy(proxy, t)
+	setNoProxy(othernoproxy, t)
+	var propertiesFileConfig = map[string]string{
+		"artifactory.resolve.contextUrl": "http://some.url.com",
+		"artifactory.publish.contextUrl": "http://some.other.url.com",
+		"artifactory.deploy.build.name":  "buildName",
+	}
+	createSimplePropertiesFile(t, propertiesFileConfig)
+	setProxy(proxyOrg, t)
+	setNoProxy(noproxyOrg, t)
+}
+
```

Comment:
This test has the proxy and the no proxy not matching and should return proxy/port